backend/assignments/1/src/main/java/backend/assignment/Coin.java

```diff
@@ -0,0 +1,49 @@
+package backend.assignment;
+
+public class Coin {
+    private int rank ;
+    private String name;
+    private String symbol;
+    private double price;
+    private long circulatingSupply;
+
+    Coin(int rank,String name,String symbol,double price,long circulatingSupply){
+        this.rank = rank;
+        this.name = name;
+        this.symbol = symbol;
+        this.price = price;
+        this.circulatingSupply = circulatingSupply;
+    }
+    // Getters
+    public int getRank() {
+        return rank;
+    }
+    public String getName() {
+        return name;
+    }
+    public String getSymbol() {
+        return symbol;
+    }
+    public synchronized double getPrice() {
```

Comment:
What if user1 needs to work on Coin1 and user2 on coin2.
But user2 will still be in wait state untill the method reopoens lock.
Can the locking be in a better place?